mode/sql/sql.js

```diff
@@ -60,10 +60,14 @@ CodeMirror.defineMode("sql", function(config, parserConfig) {
       state.tokenize = tokenLiteral(ch);
       return state.tokenize(stream, state);
     } else if ((((support.nCharCast && (ch == "n" || ch == "N"))
+        || (support.escapeConstant && (ch == "e" || ch == "E"))
         || (support.charsetCast && ch == "_" && stream.match(/[a-z][a-z0-9]*/i)))
-        && (stream.peek() == "'" || stream.peek() == '"'))) {
+        && (stream.peek() == "'" || (stream.peek() == '"' && support.doubleQuote)))) {
       // charset casting: _utf8'str', N'str', n'str'
       // ref: http://dev.mysql.com/doc/refman/5.5/en/string-literals.html
+      // escape constant: E'str', e'str'
+      // ref: https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS-ESCAPE
+      state.escapeConstSet = true;
```

Comment:
Two more questions:

 - Is it intentional that this is also being set for other types of string prefixes like `n` and `_`? It looks like it should only be relevant for `e`.

 - You don't actually need a new state field, since the state already has a `tokenize` callback that you could replace with a tokenizer function that can parse escaped strings. You could add another argument to `tokenLiteral` that determines backslash behavior, and set `tokenize` to a function that consumes the quote and then replaces itself with `tokenLiteral(quote, ...)` here.

Reply:
> Two more questions:
> 
> * Is it intentional that this is also being set for other types of string prefixes like `n` and `_`? It looks like it should only be relevant for `e`.
Oh yes, you're right. Will fix this. :/
> * You don't actually need a new state field, since the state already has a `tokenize` callback that you could replace with a tokenizer function that can parse escaped strings. You could add another argument to `tokenLiteral` that determines backslash behavior, and set `tokenize` to a function that consumes the quote and then replaces itself with `tokenLiteral(quote, ...)` here.
I didn't want to repeat the logic of tokenLiteral and wanted to reuse it. I'll try if can avoid the new state field.

Thanks for your time.

Reply:
> I didn't want to repeat the logic of tokenLiteral and wanted to reuse it.

That's what I mean—if you give it an extra argument, you can reuse it.

Reply:
Cool. Let me try it !! Thanks.

Reply:
Hi,
I tried to do it without escapeConstSet var, but the problem is, it is before the string stream starts. So I have to tell the tokenLiteral function somehow that the previous char was a escape constant. I can do it in escape constant check, but then I lose the formatting for escape constatnt.

Reply:
After seeing the `e` prefix, you can set `state.tokenize` to a tokenizer function that's specific to that situation, which starts tokenizing an escape-supporting string when it is called.

Reply:
Hi @marijnh,
This looks a lot better now. :)
Sorry for trouble.